crates/valence_server/src/layer/chunk/loaded.rs

```diff
@@ -295,6 +293,72 @@ impl LoadedChunk {
         self.assert_no_changes();
     }
 
+    /// Generates the `MOTION_BLOCKING` heightmap for this chunk, which stores
+    /// the height of the highest non motion-blocking block in each column.
+    ///
+    /// The lowest value of the heightmap is 0, which means that there are no
+    /// motion-blocking blocks in the column. In this case, rain will fall
+    /// through the void and there will be no rain particles.
+    ///
+    /// A value of 1 means that rain particles will appear at the lowest
+    /// possible height y=-64, but note that blocks cannot be placed at y=-65.
```

Comment:
Worth pointing out that -64 is not a hardcoded value. It's known as `min_y` in the "dimension type" and can be configured to go lower or higher.